jrnl/plugins/pelican_markdown_exporter.py

```diff
@@ -0,0 +1,139 @@
+        heading = "#"
+        previous_line = ""
+        warn_on_heading_level = False
+        for line in body.splitlines(True):
+            if re.match(r"^#+ ", line):
+                """ATX style headings"""
+                newbody = newbody + previous_line + heading + line
+                if re.match(r"^#######+ ", heading + line):
+                    warn_on_heading_level = True
+                line = ""
+            elif re.match(r"^=+$", line.rstrip()) and not re.match(
+                r"^$", previous_line.strip()
+            ):
+                """Setext style H1"""
+                newbody = newbody + heading + "# " + previous_line
+                line = ""
+            elif re.match(r"^-+$", line.rstrip()) and not re.match(
+                r"^$", previous_line.strip()
+            ):
+                """Setext style H2"""
+                newbody = newbody + heading + "## " + previous_line
+                line = ""
+            elif multi_tag_regex.match(line):
+                """Tag only lines"""
+                line = ""
+            else:
+                newbody = newbody + previous_line
+            previous_line = line
+        newbody = newbody + previous_line  # add very last line
+
+        # make sure the export ends with a blank line
+        if previous_line not in ["\r", "\n", "\r\n", "\n\r"]:
+            newbody = newbody + os.linesep
+
+        if warn_on_heading_level is True:
+            print(
+                "{}WARNING{}: Headings increased past H6 on export - {} {}".format(
+                    WARNING_COLOR, RESET_COLOR, date_str, entry.title
+                ),
+                file=sys.stderr,
+            )
+
+        dayone_attributes = ""
+        if hasattr(entry, "uuid"):
+            dayone_attributes += "uuid: " + entry.uuid + "\n"
+        if (
+            hasattr(entry, "creator_device_agent")
+            or hasattr(entry, "creator_generation_date")
+            or hasattr(entry, "creator_host_name")
+            or hasattr(entry, "creator_os_agent")
+            or hasattr(entry, "creator_software_agent")
+        ):
+            dayone_attributes += "creator:\n"
+            if hasattr(entry, "creator_device_agent"):
+                dayone_attributes += f"    device agent: {entry.creator_device_agent}\n"
+            if hasattr(entry, "creator_generation_date"):
+                dayone_attributes += "    generation date: {}\n".format(
+                    str(entry.creator_generation_date)
+                )
+            if hasattr(entry, "creator_host_name"):
+                dayone_attributes += f"    host name: {entry.creator_host_name}\n"
+            if hasattr(entry, "creator_os_agent"):
+                dayone_attributes += f"    os agent: {entry.creator_os_agent}\n"
+            if hasattr(entry, "creator_software_agent"):
+                dayone_attributes += (
+                    f"    software agent: {entry.creator_software_agent}\n"
+                )
```

Comment:
It would be great if this exporter didn't have to know anything about the specifics of the Day One Classic format so we can get a bit closer to the single responsibility principle. For instance, the DayOne class could keep an inventory of its metadata, and this exporter could just ask it for that metadata.

Reply:
The problem with that logic (asking the DayOne format for the metadata) is that it still needs to be presented in an export specfic format; i.e. there is no way (at least that I know of) to generalize this to a function that will work across the YAML, JSON, and XML exporters.

In any case, this is the code as it has been since 2014, and significant refactoring of the plugin is beyond the intended scope of this PR.